File: array/MaximumSubarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const maxSubArray = nums => {
  if (!nums.length) {
    return 0;
  }

  let maxValue = nums[0];
  let prevValue = nums[0];
  for (let i = 1; i < nums.length; i++) {
    // If the previous value is more than 0, add it
    // Otherwise its a deadweight, just evaluate current number
    prevValue = nums[i] + (prevValue > 0 ? prevValue : 0);
    maxValue = Math.max(prevValue, maxValue);
  }

  return maxValue;
};

File: array/MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        largest, previous = nums[0], nums[0]
        for index in range(1, len(nums)):
            # Keep adding, if 0 or negative, drop immediately
            previous = nums[index] + max(0, previous)
            largest = max(previous, largest)
        return largest